packages/runner-ct/src/iframe/iframes.tsx

```diff
@@ -22,134 +21,133 @@ export function getSpecUrl ({ namespace, spec }, prefix = '') {
 interface IFramesProps {
   state: State
   eventManager: typeof eventManager
-  config: any
+  config: Cypress.RuntimeConfigOptions
 }
 
-@observer
-export default class Iframes extends Component<IFramesProps> {
-  _disposers = []
-  containerRef = null
-  autIframe: AutIframe
-  iframeModel: IframeModel
-
-  render () {
-    const { viewportHeight, viewportWidth, scriptError, scale, screenshotting } = this.props.state
-
-    return (
-      <div
-        style={{
-          display: this.props.state.screenshotting ? 'inherit' : 'grid',
-        }}
-        className={cs('iframes-ct-container', {
-          'has-error': !!scriptError,
-          'iframes-ct-container-screenshotting': screenshotting,
-        })}
-      >
-        <div
-          ref={(container) => this.containerRef = container}
-          className={
-            cs('size-container', {
-              [styles.noSpecAut]: !this.props.state.spec,
-            })
-          }
-          style={{
-            height: viewportHeight,
-            width: viewportWidth,
-            transform: `scale(${screenshotting ? 1 : scale})`,
-          }}
+export const Iframes: React.FC<IFramesProps> = namedObserver('Iframes', (props) => {
+  const containerRef = useRef<HTMLDivElement>(null)
+  const [autIframe] = useState(new AutIframe(props.config))
+  const [disposers, setDisposers] = useState<IReactionDisposer[]>([])
+
+  const [iframeModel] = useState(
+    new IframeModel({
+      state: props.state,
+      restoreDom: autIframe.restoreDom,
+      highlightEl: autIframe.highlightEl,
+      detachDom: autIframe.detachDom,
+      snapshotControls: (snapshotProps) => (
+        <SnapshotControls
+          eventManager={props.eventManager}
+          snapshotProps={snapshotProps}
+          state={props.state}
+          onToggleHighlights={_toggleSnapshotHighlights}
+          onStateChange={_changeSnapshotState}
         />
-        <ScriptError error={scriptError} />
-        <div className='cover' />
-      </div>
-    )
-  }
-
-  componentDidMount () {
-    const config = this.props.config
-
-    this.autIframe = new AutIframe(config)
+      ),
+    }),
+  )
 
-    this.props.eventManager.on('visit:failed', this.autIframe.showVisitFailure)
-    this.props.eventManager.on('before:screenshot', this.autIframe.beforeScreenshot)
-    this.props.eventManager.on('after:screenshot', this.autIframe.afterScreenshot)
-    this.props.eventManager.on('script:error', this._setScriptError)
+  useEffect(() => {
+    props.eventManager.on('visit:failed', autIframe.showVisitFailure)
```

Comment:
Ideally I think all of these would be registered from within `AutIframe`

Reply:
I think so, too. That said, since this PR is already making a lot of changes to a fairly important part of the code, and moving them to `AutIframe` would be more than just a simple clean up (basic goal of this PR), I'd prefer to do this later (if we decide to do it at all).